test/util/genesis/legacy_genesis.go

```diff
@@ -78,6 +79,18 @@ func DocumentLegacyBytes(
 		return nil, fmt.Errorf("adding time_iota_ms field: %w", err)
 	}
 
+	// the version field used to be at this consensus_params.version.app_version
+	bz, err = maps.SetField(bz, "consensus_params.version.app_version", strconv.Itoa(int(cp.Version.App)))
+	if err != nil {
+		return nil, fmt.Errorf("adding app_version field: %w", err)
+	}
+
+	// remove the version from the new location.
+	bz, err = maps.RemoveField(bz, "consensus_params.version.app")
```

Comment:
to confirm my understanding, the `app_version` field was renamed to `app` and since this test might start using a v3 app version, we need to use the old name, right?

if so, shouldn't we keep both in case the test was started using v4 app?